packages/typescript-estree/src/node-utils.ts

```diff
@@ -335,13 +335,20 @@ export function isJSXToken(node: ts.Node): boolean {
  */
 export function getDeclarationKind(
   node: ts.VariableDeclarationList,
-): 'const' | 'let' | 'var' {
+): 'const' | 'let' | 'var' | 'using' | 'await using' {
```

Comment:
[Question] What do you think about making a dedicated type for this? Just as a convenience point, not a blocker for the PR.

Reply:
Looks good!